poly/src/polynomial/univariate/sparse.rs

```diff
@@ -343,6 +371,30 @@ mod tests {
         SparsePolynomial::from_coefficients_vec(coeffs)
     }
 
+    #[test]
+    fn test_valid_coefficients() {
+        let mut coeffs = vec![(1, Fr::from(2)), (2, Fr::from(3)), (3, Fr::from(4))];
+        let validation = SparsePolynomial::is_valid_coefficients_vec(&mut coeffs);
+        assert!(validation.is_ok());
+
+        let poly = SparsePolynomial::from_coefficients_vec(coeffs);
+        assert_eq!(poly.coeffs.len(), 3);
+    }
+
+    #[test]
+    fn test_invalid_zero_coefficient() {
+        let mut coeffs = vec![(1, Fr::from(0)), (2, Fr::from(3))];
+        let validation = SparsePolynomial::is_valid_coefficients_vec(&mut coeffs);
+        assert!(matches!(validation, Err(CoeffVecError::ZeroCoefficient)));
+    }
+
+    #[test]
+    fn test_invalid_duplicate_degrees() {
+        let mut coeffs = vec![(1, Fr::from(2)), (1, Fr::from(3))];
```

Comment:
```suggestion
        let mut coeffs = vec![(3, Fr::from(2)), (1, Fr::from(2)), (2, Fr::from(2)), (1, Fr::from(3))];
```

This way you check if the sorting works as expected.